mitodl/odl-video-service
# Remove usage of global default edx endpoint (#966)

#### Pre-Flight checklist


#### What are the relevant tickets?
Fixes #966 

#### What's this PR do?
- Adds a migration to generate `CollectionEdxEndpoint` for collections with `edx_course_id` based on the current global default endpoint. This migration also unsets that endpoint as the global default.
- Removes the global default option from django admin
- Ensures not more than endpoint per collection can be configured through django admin
- Adds new encrypted fields for `secret_key` and `client_id` into the `EdxEndpoint` data model
- Moves usages of those values from environment variable to the data model

#### How should this be manually tested?
- Ensure you have a global default endpoint and some collections with `edx_course_id` and some without
- Run the migrations
- Ensure only the collections with `edx_course_id` got `CollectionEdxEndpoint`
- Verify you can have at most 1 endpoint per collection
- Verify you can configure`secret_key` and `client_id` on `EdxEndpoint` and that these values are used
